app/src/main/java/de/danoeh/antennapod/fragment/swipeactions/SwipeActions.java

```diff
@@ -0,0 +1,282 @@
+        //reload as settings might have changed
+        fragment.getLifecycle().addObserver((LifecycleEventObserver) (source, event) -> {
+            if (event == Lifecycle.Event.ON_RESUME) {
+                resetItemTouchHelper();
+            }
+        });
+
+        return this;
+    }
+
+    public void itemTouchHelper() {
+        itemTouchHelper = new ItemTouchHelper(newSwipeCallback.construct());
+    }
+
+    private static List<SwipeAction> getPrefs(Context context, String tag, String defaultActions) {
+        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        String prefsString = prefs.getString(PREF_SWIPEACTIONS + tag, defaultActions);
+
+        ArrayList<SwipeAction> actions = new ArrayList<>();
+
+        if (!prefsString.isEmpty()) {
+            String[] rightleft = prefsString.split(",");
+
+            for (String s : rightleft) {
+                actions.add(Stream.of(swipeActions).filter(a -> a.id().equals(s)).single());
+            }
+        }
+
+        return actions;
+    }
+
+    private static List<SwipeAction> getPrefs(Context context, String tag) {
+        return getPrefs(context, tag, "");
+    }
+
+    public static List<SwipeAction> getPrefsWithDefaults(Context context, String tag) {
+        String defaultActions;
+        switch (tag) {
+            /*case InboxFragment.TAG:
+                defaultActions = new int[] {ADD_TO_QUEUE, MARK_UNPLAYED};
+                break;*/
+            case QueueFragment.TAG:
+                defaultActions = SwipeAction.REMOVE_FROM_QUEUE + "," + SwipeAction.REMOVE_FROM_QUEUE;
+                break;
+            default:
+            case EpisodesFragment.TAG:
+                defaultActions = SwipeAction.MARK_FAV + "," + SwipeAction.START_DOWNLOAD;
+                break;
+        }
+
+        return getPrefs(context, tag, defaultActions);
+    }
+
+    public static Boolean getNoActionPref(Context context, String tag) {
+        SharedPreferences prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
+        return prefs.getBoolean(PREF_NO_ACTION + tag, false);
+    }
+
+    private Boolean getNoActionPref() {
+        return getNoActionPref(fragment.requireContext(), tag);
+    }
+
+    public void resetItemTouchHelper() {
+        //refresh itemTouchHelper after prefs changed
+        if (itemTouchHelper != null) {
+            itemTouchHelper.attachToRecyclerView(null);
+        }
+        itemTouchHelper();
+        itemTouchHelper.attachToRecyclerView(recyclerView);
+    }
+
+    public interface NewSwipeCallback {
+        SimpleSwipeCallback construct();
+    }
+
+    public void setNewSwipeCallback(NewSwipeCallback newSwipeCallback) {
+        this.newSwipeCallback = newSwipeCallback;
+        itemTouchHelper();
+    }
+
+    private void showDialog(SwipeActionsDialog.Callback prefsChanged) {
+        new SwipeActionsDialog(fragment.requireContext(), tag).show(prefsChanged);
```

Comment:
Do we really need a wrapper method that is called one single time and contains one single line?

Reply:
maybe not